cub/block/block_discontinuity.cuh

```diff
@@ -182,15 +182,15 @@ private:
             T                       (&preds)[ITEMS_PER_THREAD],         ///< [out] Calling thread's predecessor items
             FlagOp                  flag_op)                            ///< [in] Binary boolean flag predicate
         {
-            preds[ITERATION] = input[ITERATION - 1];
-
-            flags[ITERATION] = ApplyOp<FlagOp>::FlagT(
-                flag_op,
-                preds[ITERATION],
-                input[ITERATION],
-                (linear_tid * ITEMS_PER_THREAD) + ITERATION);
-
-            Iterate<ITERATION + 1, MAX_ITERATIONS>::FlagHeads(linear_tid, flags, input, preds, flag_op);
+            #pragma unroll
```

Comment:
Rather than modify `Iterate`, let's remove it entirely and just use unrolled loops at the callsites.

Reply:
I had thought of the same. But seemed like copying a lot of same code. Do we want that? Thanks!

Reply:
Agreed, this provides some additional abstractions in addition to the loop, so I'm fine leaving it in. Removing the recursive instantiation should still avoid most of the compile-time overhead.